### ISSUE TITLE
OpenStudio Application crashes when making new plan in geometry editor tab

### ISSUE THREAD

**Issue by DavidGoldwasser** [NREL/OpenStudio#3502]
This happens as soon as I click on the "New" button in the Editor sub-tab in the Geometry tab

**Comment by DavidGoldwasser** [comment]
I tried opening a model made in 270 with FloorSpaceJS. When I went to editor it didn't crash right away, but then crashed a few seconds later.

**Comment by macumber** [comment]
Is this Mac only?

**Comment by DavidGoldwasser** [comment]
yes

**Comment by jmarrec** [comment]
I can reproduce with current develop on macOS 10.13.6.

**Comment by jmarrec** [comment]
Hum I think I finally found a hint after much trial and error.

in `openstudiocore/src/openstudio_lib/library/embeddable_geometry_editor.html`

I started by deleting the custom <script> around line 257. That would still crash. I then had to figure out what else was making it crash, and finally found that in the style section arouhd line 7 you have a big minified line in which there's basically:

```
@import url(https://fonts.googleapis.com/css?family=Roboto);

body,
html {
    color: #bbc3c7;
    font-size: 16px;
    font-family: Roboto, sans-serif;
    height: 100vh;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    margin: 0
}
```

If I remove the Roboto stuff, suddenly I don't get the crash anymore.

```
/* @import url(https://fonts.googleapis.com/css?family=Roboto); */

body,
html {
    color: #bbc3c7;
    font-size: 16px;
    /*font-family: Roboto, sans-serif; */
    height: 100vh;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    margin: 0
}
```

I then re-enabled the custom <script> block. Added breakpoint on FloorspaceEditor::loadEditor. Going line by line I see it go until the end. But crash happens anyways. Looks like some kind of race condition somewhere... I tried increasing the processEvents delay to 1000 instead of 200, didn't help. The Qt remote debugging console is absolutely no use here

**Comment by jmarrec** [comment]
Between the javascript and the debugger breakpoint (and a bit of luck in juddling the two) I have gotten the floorspaceJS editor to load up and render  - then it crashed anyway - so I think it may be a race condition indeed.

I had two breakpoints on FloorspaceEditor::loadEditor and FloorspaceEditor::FloorspaceEditor

I wrapped part of the javascript init function inside a setTimeout.
```
    delete window.startApp;

    console.log('Sleeping for a bit so you have time to attach to the page');
setTimeout(
      function() {
    console.log('dispatch actions');

    // don't dispatch actions until the application and data store are instantiated
    window.application.$store.dispatch('project/setUnits', { units: window.api.config.units, editable: window.api.config.unitsEditable });
    window.application.$store.dispatch('project/setShowImportExport', window.api.config.showImportExport);
    window.application.$store.dispatch('project/setSpacing', { spacing: window.api.config.initialGridSize });
    window.application.$store.dispatch('project/setNorthAxis', { north_axis: window.api.config.initialNorthAxis });

    window.application.$store.dispatch('project/setMapEnabled', { enabled: window.api.config.showMapDialogOnStart });
    window.application.$store.dispatch('project/setMapVisible', { visible: window.api.config.showMapDialogOnStart });

    window.application.$store.dispatch('project/setMapLatitude', { latitude: window.api.config.defaultLocation.latitude });
    window.application.$store.dispatch('project/setMapLongitude', { longitude: window.api.config.defaultLocation.longitude });

    window.application.$store.dispatch('application/setCurrentSnapMode', { snapMode: window.api.config.snapMode });

    console.log('window.api.config');
    window.api.initAlreadyRun = true;
      }, 9000);
```

**Comment by macumber** [comment]
@jmarrec should this be assigned to @kbenne or are you able to keep looking at it

**Comment by jmarrec** [comment]
I'm kinda at a loss on this one, and with my old mac it's not practical to try more stuff like building Qt in Debug, so if Kyle can take a look that would be great.

**Comment by macumber** [comment]
Related to errors in SSL connections, possible fix in https://github.com/NREL/OpenStudio/pull/3552

**Comment by tijcolem** [comment]
I ran in debug and produced the same call stack as @jmarrec .  It points to an issue with the method in QtNetwork::ssl3_connect which is in the Qt shared library that is loaded at run time. This dylib is included in the version of [Qt](https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/CMakeLists.txt#L973) that is statically linking OpenSSL 1.02q vs the previous version that used the OSX system SSL library 0.9.8. 

I rebuilt using the previous version of Qt that links system 0.9.8 and now the app is loading https assets and not crashing, so clearly the issue stems from that version of Qt.  It's not entirely clear why that dylib is segfaulting ( maybe the linker didn't get passed correct flags at build time).  

I think we can safely revert back to using the previous version as the openstudio cli uses the static qt libraries vs dynamic which works fine. The issue reverting back is loading https assets from servers that no longer support tls < 1.1.   However, even in these situations, I would expect Qt to catch the unsupported protocol vs a run time error, which is much improved over the current situation. 

I can put these changes in the CMake config, rebuild and run regression tests to make sure this proposed change doesn't affect anything else within the app. We can revisit the Qt shared build and determine the root cause.
